tests/src/whisk/core/database/test/CouchDbRestClientTests.scala

```diff
@@ -75,48 +87,96 @@ class CouchDbRestClientTests extends FlatSpec
+
+        val proxyPort = 15975
+        val proxyActor = actorSystem.actorOf(Props(new RESTProxy("0.0.0.0", proxyPort)(dbAuthority, config.dbProtocol == "https")))
+
+        val proxiedClient = new ExtendedCouchDbRestClient("http", "localhost", proxyPort, config.dbUsername, config.dbPassword, dbName)
+
         // sprays the client with requests, makes sure they are all answered
         // despite temporary connection failure.
         val numRequests = 30
         val timeSpan = 5.seconds
         val delta = timeSpan / numRequests
 
-        val promises = Vector.fill(numRequests)(Promise[Either[StatusCode,JsObject]])
+        val promises = Vector.fill(numRequests)(Promise[Either[StatusCode, JsObject]])
 
-        for(i <- 0 until numRequests) {
-            actorSystem.scheduler.scheduleOnce(delta * (i+1)) {
-                client.instanceInfo().andThen({ case r => promises(i).tryComplete(r) })
+        for (i <- 0 until numRequests) {
+            actorSystem.scheduler.scheduleOnce(delta * (i + 1)) {
+                proxiedClient.instanceInfo().andThen({ case r => promises(i).tryComplete(r) })
             }
         }
 
         // Mayhem! Havoc!
         actorSystem.scheduler.scheduleOnce(2.5.seconds, proxyActor, RESTProxy.UnbindFor(1.second))
 
         // What a type!
-        val futures: Vector[Future[Try[Either[StatusCode,JsObject]]]] =
+        val futures: Vector[Future[Try[Either[StatusCode, JsObject]]]] =
             promises.map(_.future.map(e => Success(e)).recover { case t: Throwable => Failure(t) })
 
         whenReady(Future.sequence(futures), Timeout(timeSpan * 2)) { results =>
             // We check that the first result was OK
             // (i.e. the service worked before the disruption)
             results.head.toOption shouldBe defined
-            checkResponse(results.head.get)
+            checkInstanceInfoResponse(results.head.get)
 
             // We check that the last result was OK
             // (i.e. the service worked again after the disruption)
             results.last.toOption shouldBe defined
-            checkResponse(results.last.get)
+            checkInstanceInfoResponse(results.last.get)
 
             // We check that there was at least one error
             // (i.e. we did manage to unbind for a while)
             results.find(_.isFailure) shouldBe defined
         }
     }
+
+    it should "upload then download an attachment" in {
+        assume(config.dbProvider == "Cloudant" || config.dbProvider == "CouchDB")
+
+        val docId = "some_doc"
+        val doc = JsObject("greeting" -> JsString("hello"))
+        val attachmentName = "misc"
+        val attachmentType = ContentTypes.`text/plain(UTF-8)`
+        val attachment = ("""
+            | There once was an OpenWhisk action
+            | Without an exec as companion
+            |   It's not in the document
+            |   It is an attachment!
+            | Stream it for fast invocation.
+        """).stripMargin
```

Comment:
There once was an action named foo,
With no exec, he didn't know what to do!
   But don't panic my friend,
   Just stream it till the end!
And all's well and good.   Phew!